python/rainbow/simulators/prox_soft_bodies/collision_detection.py

```diff
@@ -295,30 +305,34 @@ def _contact_reduction(engine, stats, debug_on):
     contact point information. One may think of this as a kind of post-filtering process to clean up the
      contact point information.
 
-     :param engine:      The current engine instance we are working with.
-     :param stats:       A dictionary where to add more profiling and timing measurements.
-     :param debug_on:    Boolean flag for toggling debug (aka profiling) info on and off.
-     :return:            A dictionary with profiling and timing measurements.
+    Args:
+        engine (Engine): The current engine instance we are working with.
+        stats (dict):  A dictionary where to add more profiling and timing measurements.
+        debug_on (bool): Boolean flag for toggling debug (aka profiling) info on and off.
+
+    Returns:
+        dict: A dictionary with profiling and timing measurements.
     """
     reduction_timer = None
     if debug_on:
         reduction_timer = Timer("contact_point_reduction", 8)
         reduction_timer.start()
-    # TODO 2020-09-07 Kristian: This brute force implementation can be implemented better
+
+    seen_ids = set()
     reduced_list = []
-    for cp1 in engine.contact_points:
-        unique = True
-        for cp2 in reduced_list:
-            if {cp1.bodyA, cp1.bodyB} == {cp2.bodyA, cp2.bodyB} and (
-                cp1.p == cp2.p
-            ).all():
-                unique = False
-        if unique:
-            reduced_list.append(cp1)
+
+    for cp in engine.contact_points:
+        uid = _cp_unique_id(cp)
```

Comment:
This function call is likely expensive in Python. Why not just have the "tuple" creation done inline and add an inline code comment that this creates a unique cp id?